tls/extensions/s2n_server_key_share.c

```diff
@@ -93,8 +93,11 @@ static int s2n_server_key_share_recv(struct s2n_connection *conn, struct s2n_stu
     S2N_ERROR_IF(supported_curve_index < 0, S2N_ERR_ECDHE_UNSUPPORTED_CURVE);
 
     /* If this is a HelloRetryRequest, we won't have a key share. We just have the selected group.
-     * Exit early so a proper keyshare can be generated. */
-    if (s2n_is_hello_retry_required(conn)) {
+     * Set the server negotiated curve and exit early so a proper keyshare can be generated. 
+     */
+    if (s2n_is_hello_retry_handshake(conn)) {
+        struct s2n_ecc_evp_params* server_ecc_evp_params = &conn->secure.server_ecc_evp_params;
+        server_ecc_evp_params->negotiated_curve = ecc_pref->ecc_curves[supported_curve_index];
```

Comment:
Should we just move the "set server curve" code form lines 107-108 to before the s2n_is_hello_retry_handshake check, instead of duplicating it here?